runner/app/live/Sam2Wrapper/wrapper.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import List, Optional, Tuple
+from PIL import Image
+import torch
+from sam2.build_sam import build_sam2_camera_predictor
+from omegaconf import OmegaConf
+from hydra.utils import instantiate
+from hydra import initialize_config_dir, compose
+from hydra.core.global_hydra import GlobalHydra
+import os
+
+MODEL_MAPPING = {
+    "facebook/sam2-hiera-tiny": {
+        "config": "sam2_hiera_t.yaml",
+        "checkpoint": "sam2_hiera_tiny.pt"
+    },
+    "facebook/sam2-hiera-small": {
+        "config": "sam2_hiera_s.yaml",
+        "checkpoint": "sam2_hiera_small.pt"
+    },
+    "facebook/sam2-hiera-large": {
+        "config": "sam2_hiera_l.yaml",
+        "checkpoint": "sam2_hiera_large.pt"
+    }
+}
+
+torch.autocast(device_type="cuda", dtype=torch.bfloat16).__enter__()
+if torch.cuda.get_device_properties(0).major >= 8:
+    torch.backends.cuda.matmul.allow_tf32 = True
+    torch.backends.cudnn.allow_tf32 = True
+
+# Initialize Hydra to load the configuration
+if GlobalHydra.instance().is_initialized():
+    GlobalHydra.instance().clear()
+
+class Sam2Wrapper:
+    def __init__(
+        self,
+        model_id_or_path: str,
+        device: Optional[str] = None,
+        **kwargs
+    ):
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Moved to `segment_anything_2.py` and passed device type to hydra config. 

I wanted to import `get_torch_device` from `app.pipelines.utils`, but due to how `infer.py` is initialized from `app.pipelines.live`, it didn't seem worthwhile to add absolute path imports to do so

Reply:
`infer.py` is a separate application it should only use code from within the `live/` folder!

Reply:
What we are missing I think is this: https://github.com/livepeer/ai-worker/blob/e965eb7276261734d6565a025dee524fcb469c49/runner/app/pipelines/live_video_to_video.py#L46
(then passing it internally on `infer.py` to the pipelines)

Reply:
Can we handle this on a separate PR? I briefly reviewed the code path and think it would be better to handle separately since it affects all pipelines

Reply:
Definitely! It was just a reminder of that existing `TODO`.